In [2]:
import numpy as np

Question1- If A=[1 2 3,2 3 -1,-3 1 2]
                     B=[1 0 2,0 1 2,1 2 0] these both are 3*3 matrix, obtain the product AB and
                     BA and show that AB is not equal to BA.


In [3]:
A = np.array([[1,2,3],[2,3,-1],[-3,1,2]])
A

array([[ 1,  2,  3],
       [ 2,  3, -1],
       [-3,  1,  2]])

In [4]:
B=np.array([[1, 0, 2],[0, 1, 2],[1, 2, 0]])
B

array([[1, 0, 2],
       [0, 1, 2],
       [1, 2, 0]])

In [5]:
AB = np.dot(A,B)
AB

array([[ 4,  8,  6],
       [ 1,  1, 10],
       [-1,  5, -4]])

In [6]:
BA = np.dot(B,A)
BA

array([[-5,  4,  7],
       [-4,  5,  3],
       [ 5,  8,  1]])

In [7]:
check = (AB == BA).all()
check

False

Question2-If A=[3 1,-1 2] is 2*2 matrix, show that (A*A) - 5A + 7 = 0.


In [8]:
A=np.array([[3,1],[-1 ,2]])
A

array([[ 3,  1],
       [-1,  2]])

In [9]:
AA = np.dot(A,A)
AA

array([[ 8,  5],
       [-5,  3]])

In [10]:
I = np.array([[1,0],[0,1]])
I

array([[1, 0],
       [0, 1]])

In [11]:
x5A = np.dot(5,A)
x5A

array([[15,  5],
       [-5, 10]])

In [12]:
x7I = np.dot(7,I)
x7I

array([[7, 0],
       [0, 7]])

In [13]:
AAx5A = np.subtract(AA,x5A)
AAx5A

array([[-7,  0],
       [ 0, -7]])

In [14]:
AAx5Ax7 = np.add(AAx5A,x7I)
AAx5Ax7

array([[0, 0],
       [0, 0]])

Question3-If A=[3 2 1,-5 0 6] and B=[-4 -5 -2, 3 1 8], verify that (A+B)’=A’+B’
                  (where A’ means transpose of A).


In [15]:
A=np.array([[3, 2, 1],[-5, 0 ,6]])
A

array([[ 3,  2,  1],
       [-5,  0,  6]])

In [16]:
B=np.array([[-4, -5, -2],[ 3, 1, 8]])
B

array([[-4, -5, -2],
       [ 3,  1,  8]])

In [17]:
AB = np.add(A,B)
AB

array([[-1, -3, -1],
       [-2,  1, 14]])

In [18]:
AB_transpose = np.transpose(AB)
AB_transpose

array([[-1, -2],
       [-3,  1],
       [-1, 14]])

In [19]:
A_transpose = np.transpose(A)
A_transpose

array([[ 3, -5],
       [ 2,  0],
       [ 1,  6]])

In [20]:
B_transpose = np.transpose(B)
B_transpose

array([[-4,  3],
       [-5,  1],
       [-2,  8]])

In [21]:
AB_transposee = np.add(A_transpose,B_transpose)
AB_transposee

array([[-1, -2],
       [-3,  1],
       [-1, 14]])

In [22]:
verify = (AB_transpose == AB_transposee).all()
verify

True


Question4- If A=[3 -4,1 -1], show that A - A’ is skewsymmetric, where A’ is the transpose                                   
                   of  A. 


In [23]:
A = np.array([[3,-4],[1,-1]])
A

array([[ 3, -4],
       [ 1, -1]])

In [24]:
A_transpose = np.transpose(A)
A_transpose

array([[ 3,  1],
       [-4, -1]])

In [25]:
A_A_transpose = np.subtract(A,A_transpose)
A_A_transpose

array([[ 0, -5],
       [ 5,  0]])

In [26]:
skewsymmetric = np.dot(-1,A_A_transpose)
skewsymmetric

array([[ 0,  5],
       [-5,  0]])

If A=[5 7 3,1 2 6,1 4 3]
                     B=[1 2 0,4 0 5,0 3 2] these both are 3*3 matrix, then verify that (AB)’=B’A’


In [27]:
A=np.array([[5, 7, 3],[1, 2, 6],[1, 4, 3]])
A

array([[5, 7, 3],
       [1, 2, 6],
       [1, 4, 3]])

In [28]:
B=np.array([[1 ,2, 0],[4, 0, 5],[0, 3, 2]])
B

array([[1, 2, 0],
       [4, 0, 5],
       [0, 3, 2]])

In [29]:
AB = np.dot(A,B)
AB

array([[33, 19, 41],
       [ 9, 20, 22],
       [17, 11, 26]])

In [30]:
AB_transpose = np.transpose(AB)
AB_transpose

array([[33,  9, 17],
       [19, 20, 11],
       [41, 22, 26]])

In [31]:
A_transpose = np.transpose(A)
A_transpose

array([[5, 1, 1],
       [7, 2, 4],
       [3, 6, 3]])

In [32]:
B_transpose = np.transpose(B)
B_transpose

array([[1, 4, 0],
       [2, 0, 3],
       [0, 5, 2]])

In [33]:
BA_transpose = np.dot(B_transpose,A_transpose)
BA_transpose

array([[33,  9, 17],
       [19, 20, 11],
       [41, 22, 26]])

In [34]:
verify = (AB_transpose == BA_transpose).all() 
verify

True

Question6-  find the inverse of A=[2 -1 1,3 0 -1,2 6 0]. Verify that A*A(inverse)= I  
                   (identity matrices of order 3*3)



In [35]:
A = np.array([[2,-1,1],[3,0,-1],[2,6,0]])
A

array([[ 2, -1,  1],
       [ 3,  0, -1],
       [ 2,  6,  0]])

In [36]:
A_inverse = np.linalg.inv(A)
A_inverse

array([[ 0.1875 ,  0.1875 ,  0.03125],
       [-0.0625 , -0.0625 ,  0.15625],
       [ 0.5625 , -0.4375 ,  0.09375]])

In [37]:
I = np.identity(3)
I

array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

In [38]:
AA_inverse = np.dot(A,A_inverse)
AA_inverse

array([[1.00000000e+00, 5.55111512e-17, 0.00000000e+00],
       [0.00000000e+00, 1.00000000e+00, 0.00000000e+00],
       [0.00000000e+00, 0.00000000e+00, 1.00000000e+00]])

In [39]:
AA_inv = np.round_(AA_inverse, decimals = 3)
AA_inv

array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

In [40]:
verify = (AA_inv == I).all()
verify

True

Question7- Verify that (inverse of AB) = (inverse of B)*(inverse of A) for the matrices A=[2 1,
                   5 3] and B=[4 5,3 4]

In [41]:
A=np.array([[2, 1],[ 5, 3]])
A

array([[2, 1],
       [5, 3]])

In [42]:
B = np.array([[4 ,5],[3, 4]])
B

array([[4, 5],
       [3, 4]])

In [43]:
inverse_A = np.linalg.inv(A)
inverse_B = np.linalg.inv(B)

In [44]:
AB = np.dot(A,B)
AB

array([[11, 14],
       [29, 37]])

In [45]:
inverse_AB = np.linalg.inv(AB)
inverse_AB = np.round(inverse_AB,decimals=0)
inverse_AB

array([[ 37., -14.],
       [-29.,  11.]])

In [46]:
inverse_BA = np.dot(inverse_B,inverse_A)
inverse_BA = np.round(inverse_BA,decimals=0)
inverse_BA

array([[ 37., -14.],
       [-29.,  11.]])

In [47]:
verify = (inverse_AB == inverse_BA).all()
verify

True

Question1-
 Suppose that the data for analysis includes the attribute age. The age values for the data  are (in increasing order) 13, 15, 16, 16, 19, 20, 20, 21, 22, 22, 25, 25, 25, 25, 30, 33, 33, 35, 35, 35, 35, 36, 40, 45, 46, 52, 70.
 (a) What is the mean of the data? What is the median? 
 (b) What is the mode of the data?
 (c) What is the range of the data? 
(d) Can you find the first quartile (Q1) and the third quartile (Q3) of the data?


In [48]:
age = [13, 15, 16, 16, 19, 20, 20, 21, 22, 22, 25, 25, 25, 25, 30, 33, 33, 35, 35, 35, 35, 36, 40, 45, 46, 52, 70]

In [49]:
age

[13,
 15,
 16,
 16,
 19,
 20,
 20,
 21,
 22,
 22,
 25,
 25,
 25,
 25,
 30,
 33,
 33,
 35,
 35,
 35,
 35,
 36,
 40,
 45,
 46,
 52,
 70]

In [50]:
mean = np.mean(age)
mean

29.962962962962962

In [51]:
median = np.median(age)
median

25.0

In [52]:
import statistics
mode = statistics.multimode(age)
mode

[25, 35]

In [53]:
max_value = max(age)
max_value

70

In [54]:
min_value = min(age)
min_value

13

In [55]:
range = max_value - min_value
range

57

In [56]:
first_quantile = np.quantile(age,0.25)
first_quantile

20.5

In [57]:
third_qunatile = np.quantile(age,0.75)
third_qunatile

35.0

In [58]:
IQR = third_qunatile - first_quantile
IQR

14.5

Question 3- 
A tire manufacturer wants to determine the inner diameter of a certain grade of the tire. Ideally, the diameter would be 570 mm. The data are as follows: 572, 572, 573, 568, 569, 575, 565, 570.
 (a) Find the mean and median.
 (b)  standard deviation, and range.
 (c) Using the calculated statistics in parts (a) and (b), can you comment on the quality of the tires?


In [59]:
data = [ 572, 572, 573, 568, 569, 575, 565, 570]

## Diameter of tire = 570mm 

In [60]:
tire_mean = np.mean(data)
tire_mean

570.5

In [61]:
tire_median = np.median(data)
tire_median

571.0

In [62]:
standard_deviation = statistics.stdev(data)
standard_deviation

3.1622776601683795

In [63]:
max_tire = max(data)
min_tire = min(data)

In [64]:
tire_range = max_tire - min_tire
tire_range

10

# Mean and median of the tire is more than the Diameter of the tire.

# 3.162 is deviated from the average value of inner diameter of the tire.

#  We get Standard deviation as 3.162 ,which is greater than 2 therefore there is a more spread of tire data.